### PR TITLE
Update ci-cd docs

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Update ci-cd docs

### CODE REVIEW

File: docs/contributor/04-30-ci-cd.md [hunk 1]
```diff
@@ -7,61 +7,61 @@ The following CI jobs are part of the development cycle.
...
 
 ## Pipelines Running on the Release Branch
 
 The following CI jobs generate release artifacts and verify the functional correctness of Kyma Istio Operator.
 
 | Name                                                                                                                                                                         | Description                                                                             |
 |------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------|
-| [`Istio integration test`]() | Executes the main integration test suite for the Istio module on a k3d cluster.         |
-| [`rel-istio-build`]()           | Builds Kyma Istio Operator's image on the release and pushes it to the `prod` registry. |
+| [`Istio integration test`](https://github.com/kyma-project/istio/blob/fd34803dd516c3b614fda4f5b9d9a4c8b46372bb/.github/workflows/main-integration.yaml#L46) | Executes the main integration test suite for the Istio module on a k3d cluster.         |
```
Comment (triffer) [inline docs/contributor/04-30-ci-cd.md:58]: We also run the "main integration tests" workflow on release branches: https://github.com/kyma-project/istio/blob/d91645ab9c8935da03f82ed87623663cf2a621ad/.github/workflows/main-integration.yaml#L18
The question is also why we don't do it for api-gateway.

File: docs/contributor/04-30-ci-cd.md
Comment (triffer) [inline docs/contributor/04-30-ci-cd.md:1]: We are missing the [UI Test workflow](https://github.com/kyma-project/istio/blob/d91645ab9c8935da03f82ed87623663cf2a621ad/.github/workflows/ui-tests.yaml#L23).

File: docs/contributor/04-30-ci-cd.md (hunk 1 above)
Comment (triffer) [inline docs/contributor/04-30-ci-cd.md:58]: Same as for API Gateway review, I think we should include the [release workflow](https://github.com/kyma-project/istio/blob/main/.github/workflows/create-release.yaml) as well.

File: docs/contributor/04-30-ci-cd.md
Comment (triffer) [inline docs/contributor/04-30-ci-cd.md:1]: We probably also want to document the [workflow that verifies GitHub Action versions are pinned](https://github.com/kyma-project/istio/blob/main/.github/workflows/verify-commit-pins.yaml).

**Review by triffer — CHANGES_REQUESTED** [review]

File: docs/contributor/04-30-ci-cd.md (hunk 1 above)
Comment (barchw) [reply docs/contributor/04-30-ci-cd.md:61]: I think we should do it in api-gateway

Comment (triffer) [reply docs/contributor/04-30-ci-cd.md:61]: I'll add it to my list and do it as a coffee break task once the documentation PR of api-gateway s merged.

**Review by triffer — DISMISSED** [review]

File: docs/contributor/04-30-ci-cd.md [hunk 2]
```diff
@@ -7,61 +7,67 @@ The following CI jobs are part of the development cycle.
...
+| [`Pull Request / Run unit tests`](https://github.com/kyma-project/istio/blob/fd34803dd516c3b614fda4f5b9d9a4c8b46372bb/.github/workflows/pull-request.yaml#L40)                                              | Runs unit tests with code coverage information.                                                                                                        |
+| [`UI Tests / Run UI tests`](https://github.com/kyma-project/istio/blob/d91645ab9c8935da03f82ed87623663cf2a621ad/.github/workflows/ui-tests.yaml#L22) | Runs UI tests for Busola extenstions of API Gateway module and API Rules. |
+| [`Workflow validation / Ensure SHA pinned actions`](https://github.com/kyma-project/istio/blob/d91645ab9c8935da03f82ed87623663cf2a621ad/.github/workflows/verify-commit-pins.yaml#L15) | Verifies that all external actions are pinned to a commit. |
 
-## Pipelines Running on the Main Branch
+## Pipelines Running on merge to main and all release branches
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:22]:
```suggestion
## Pipelines Running on Merge to Main and All Release Branches
```

File: docs/contributor/04-30-ci-cd.md [hunk 3]
```diff
@@ -7,61 +7,67 @@ The following CI jobs are part of the development cycle.
...
 
-The following CI jobs run on the main branch.
+The following CI jobs run on merge to main and all release branches.
```
Comment (nataliasitko) [inline docs/contributor/04-30-ci-cd.md:24]:
```suggestion
The following CI jobs run when you merge changes to the `main` branch and all release branches.
```

**Review by nataliasitko — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**